### ISSUE TITLE
Error with bar plot and no data

### ISSUE THREAD

**Issue by Acanthostega** [matplotlib/matplotlib#5237]
Hi everyone,

generating an empty bar plot with the current master branch of the repo creates an error:
```
ValueError: incompatible sizes: argument 'tick_label' must be length 0 or string
```

For the code example:
```python
import matplotlib.pyplot as plt

fig = plt.figure()
ax = fig.add_subplot(111)
ax.bar([], [])
```

With the 1.4.3 version of matplotlib from pip the code is working but not with the master branch. Is this now a normal behaviour ?

**Comment by tacaswell** [comment]
This is the SCCE fixed by #4946.

**Comment by jenshnielsen** [comment]
Closed by #5239
